src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        Storage.removeFruit(transaction.getFruit(), transaction.getQuantity());

```

Comment:
Before removing fruits, you should check if the storage contains enough fruit to be removed. If the quantity to remove is greater than the quantity available, you should throw a RuntimeException to indicate that the purchase cannot be completed due to insufficient stock.